content/docs/07-reference/01-ai-sdk-core/66-extract-reasoning-middleware.mdx

```diff
@@ -43,6 +43,13 @@ const middleware = extractReasoningMiddleware({
       description:
         'The separator to use between reasoning and text sections. Defaults to "\\n"',
     },
+    {
+      name: 'startWithReasoning',
+      type: 'boolean',
+      isOptional: true,
+      description:
+        'Prepends the response with the reasoning tag. Primarily used for patching edge cases in reasoning extraction in certain models, see [here](https://sdk.vercel.ai/docs/guides/r1#deepseek-r1-middleware) for more details. Defaults to false',
```

Comment:
I would say something like: "Starts with reasoning tokens. Set to true when the response always starts with reasoning and the initial <think> tag is omitted. Defaults to false".

(describe what it does, add use case, no docs links that are hard to maintain)

Reply:
good point! we'll need to keep an eye out on any future deepseek changes to this behavior 👀